libs/community/langchain_community/vectorstores/neo4j_vector.py

```diff
@@ -1449,6 +1449,8 @@ def from_existing_graph(
                 "LIMIT 1000"
             )
             data = store.query(fetch_query, params={"props": text_node_properties})
+            if not data:
```

Comment:
This doesn't really solve anything related to huggingface

Reply:
Ok, so I just re-ran the tests in [here](https://github.com/langchain-ai/langchain/pull/24912/files#diff-97f27032565f976999bcb3fdf1a3f44a34ac8ae47f6cb3c7ad9fa5320ecd022d) and replaced the 

`FakeHuggingFaceEmbeddings` with 
```
embeddings = HuggingFaceEmbeddings(
        model_name="sentence-transformers/all-mpnet-base-v2"
    )
```
and commented out the `if not data:` line.

I still get the following error:

```bash
langchain_community/vectorstores/neo4j_vector.py:1454: in from_existing_graph
    text_embeddings = embedding.embed_documents([el["text"] for el in data])
/home/vscode/langchain-py-env/lib/python3.11/site-packages/langchain_huggingface/embeddings/huggingface.py:87: in embed_documents
    embeddings = self.client.encode(
>               if all_embeddings[0].dtype == torch.bfloat16:
E               IndexError: list index out of range

/home/vscode/langchain-py-env/lib/python3.11/site-packages/sentence_transformers/SentenceTransformer.py:565: IndexError
```

So the issue people are running into is that when [this query](https://github.com/langchain-ai/langchain/blob/master/libs/community/langchain_community/vectorstores/neo4j_vector.py#L1451) returns an empty list, which it does when all documents have already been embedded, then HF throws an error as `embed_documents` looks for `all_embeddings[0]`, resulting in the `IndexError`.  

So not proceeding with embeddings when it's embedding already occurred results in avoiding the `IndexError` without apparent side effects.  Both the error and fix appear to align with what others have observed [here](https://github.com/langchain-ai/langchain/issues/24401).

Reply:
Then add just this line and nothing else.

Reply:
Ok, so remove the tests?

Reply:
sure

Reply:
Ok, let me give the tests another shot by using the real huggingface embeddings, as I would like to show the issue.  But if it doesn't look good, I'll just remove the tests.

Reply:
Ok, I wasn't sure how to best update the `FakeHuggingFaceEmbeddings` or if it was ok to reference use the real embeddings, so just removed the test.

Reply:
ok, now it's GTG... Thanks